### PR TITLE
[IOTDB-4742] [Consensus] add new api: addNewNodeToExistedGroup

### PR DESCRIPTION
**Pull request by SzyWilliam**
It is for RatisConsensus.

### CODE REVIEW

File: consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java [hunk 1]
```diff
@@ -127,6 +133,8 @@ public RatisConsensus(ConsensusConfig config, IStateMachine.Registry registry)
         properties, Collections.singletonList(new File(config.getStorageDir())));
     GrpcConfigKeys.Server.setPort(properties, config.getThisNodeEndPoint().getPort());
 
+    addExecutor = Executors.newSingleThreadExecutor(new IoTThreadFactory("ratis-add"));
```
Comment (JackieTien97) [inline consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java:136]:
```suggestion
    addExecutor = IoTDBThreadPoolFactory.newCachedThreadPool("ratis-add");
```

File: consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java [hunk 2]
```diff
@@ -463,6 +484,38 @@ public ConsensusGenericResponse changePeer(ConsensusGroupId groupId, List<Peer>
...
 
+  @Override
+  public ConsensusGenericResponse addNewNodeToExistedGroup(
+      ConsensusGroupId groupId, Peer newNode, List<Peer> originalGroup) {
+
+    CompletableFuture<ConsensusGenericResponse> addResp =
+        CompletableFuture.supplyAsync(() -> addPeer(groupId, newNode), addExecutor);
+
+    try {
+      Thread.sleep(500);
```
Comment (JackieTien97) [inline consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java:495]:
```suggestion
      TimeUnit.MILLISECONDS.sleep(500);
```

File: consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java [hunk 3]
```diff
@@ -151,8 +159,16 @@ public void start() throws IOException {
 
   @Override
   public void stop() throws IOException {
-    clientManager.close();
-    server.close();
+    addExecutor.shutdown();
+    try {
+      addExecutor.awaitTermination(5, TimeUnit.SECONDS);
+    } catch (InterruptedException e) {
+      logger.warn("{}: interrupted when shutting down add Executor", this);
```
Comment (yifuzhou) [inline consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java:166]: Should we add exception e to the log?

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java (hunk 3 above)
Comment (SzyWilliam) [reply consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java:166]: I'll add the exception. Thanks!

File: consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java (hunk 1 above)
Comment (SzyWilliam) [reply consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java:136]: Should we use cached thread pool here? Or just a single thread pool?

File: consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java (hunk 2 above)
Comment (SzyWilliam) [reply consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java:495]: done

File: consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java (hunk 1 above)
Comment (SzyWilliam) [reply consensus/src/main/java/org/apache/iotdb/consensus/ratis/RatisConsensus.java]: done.

**Review by CRZbulabula — APPROVED** [review]
LGTM～

File: consensus/src/main/java/org/apache/iotdb/consensus/IConsensus.java [hunk 4]
```diff
@@ -106,6 +107,17 @@ public interface IConsensus {
    */
   ConsensusGenericResponse changePeer(ConsensusGroupId groupId, List<Peer> newPeers);
 
+  /** Call this method on any member of originalGroup */
+  default ConsensusGenericResponse addNewNodeToExistedGroup(
```
Comment (Beyyes) [inline consensus/src/main/java/org/apache/iotdb/consensus/IConsensus.java:111]: Add more comments like `createPeer` and `addPeer`?

**Review by Beyyes — APPROVED** [review]

**Review by yifuzhou — APPROVED** [review]

File: consensus/src/main/java/org/apache/iotdb/consensus/IConsensus.java (hunk 4 above)
Comment (SzyWilliam) [reply consensus/src/main/java/org/apache/iotdb/consensus/IConsensus.java:123]: I've added the javadoc. Please take a look again, thanks!

**Pull request merged**